web/src/stream_popover.ts

```diff
@@ -509,6 +509,9 @@ export async function build_move_topic_to_stream_popover(
             // user does not have permission to edit topic.
             new_topic_name = new_topic_name.trim();
         }
+        if (new_topic_name?.trim() === "") {
+            new_topic_name = "(no topic)";
```

Comment:
This commit can be the second commit because it fixes a thing that was not working before, i.e. editing a topic to be empty.

Reply:
Would also be good to explain this in a commit message that it failed previously due to server rejecting empty string as topics.

Reply:
Done!